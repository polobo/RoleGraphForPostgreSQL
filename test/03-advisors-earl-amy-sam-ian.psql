-- Here we demonstrate the basic presentations for the various grant specification options
-- We've already demonstrated both set and inherit being possessed.  While admin is segregated
-- in the presentation here we do show each of three remaining combinations of set/inherit
-- with and without admin option.
-- Amy (admin only) holds just an admin option while Earl doesn't hold anything (empty)
-- Ian holds inherit
-- Sam holds set
-- There is a separate service advisor group for the admin and non-admin cases to aid in reading
-- Also for ease of reading, these are created by the superuser so only the explicit grants below are shown

\if :is_su
    RESET createrole_self_grant;
    SELECT format('CREATE ROLE u3_earl_%s LOGIN;', :'ctx') \gexec
    SELECT format('CREATE ROLE u3_ian_%s LOGIN;', :'ctx') \gexec
    SELECT format('CREATE ROLE u3_sam_%s LOGIN;', :'ctx') \gexec

    SELECT format('CREATE ROLE u3_amy_admin_%s LOGIN;', :'ctx') \gexec
    SELECT format('CREATE ROLE u3_ian_admin_%s LOGIN;', :'ctx') \gexec
    SELECT format('CREATE ROLE u3_sam_admin_%s LOGIN NOINHERIT;', :'ctx') \gexec
    SELECT format('CREATE ROLE g3_service_advisor_%s NOLOGIN;', :'ctx') \gexec

    SELECT format('GRANT g3_service_advisor_%1$s TO u3_earl_%1$s WITH SET FALSE, INHERIT FALSE, ADMIN FALSE;', :'ctx') \gexec
    SELECT format('GRANT g3_service_advisor_%1$s TO u3_ian_%1$s WITH SET FALSE, INHERIT TRUE, ADMIN FALSE;', :'ctx') \gexec
    SELECT format('GRANT g3_service_advisor_%1$s TO u3_sam_%1$s WITH SET TRUE, INHERIT FALSE, ADMIN FALSE;', :'ctx', :'ctx') \gexec

    SELECT format('CREATE ROLE g3_advisor_admin_%s NOLOGIN;', :'ctx') \gexec

    SELECT format('GRANT g3_advisor_admin_%1$s TO u3_amy_admin_%1$s WITH SET FALSE, INHERIT FALSE, ADMIN OPTION;', :'ctx') \gexec
    SELECT format('GRANT g3_advisor_admin_%1$s TO u3_ian_admin_%1$s WITH SET FALSE, INHERIT TRUE, ADMIN OPTION;', :'ctx') \gexec
    SELECT format('GRANT g3_advisor_admin_%1$s TO u3_sam_admin_%1$s WITH SET TRUE, INHERIT FALSE, ADMIN OPTION;', :'ctx') \gexec

\endif
