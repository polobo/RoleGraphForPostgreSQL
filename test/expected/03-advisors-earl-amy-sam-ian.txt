                                                                      Test Result for 03-advisors-earl-amy-sam-ian
        rolname        |                administration                 | memberof_users |             memberof_groups              |            member_users            | member_groups 
-----------------------+-----------------------------------------------+----------------+------------------------------------------+------------------------------------+---------------
 g3_advisor_admin_su   | by u3_amy_admin_su from superuser            +|                |                                          | u3_ian_admin_su from superuser (i)+| 
                       | by u3_ian_admin_su from superuser            +|                |                                          | u3_sam_admin_su from superuser (s) | 
                       | by u3_sam_admin_su from superuser             |                |                                          |                                    | 
 g3_service_advisor_su | empty by u3_earl_su from superuser            |                |                                          | u3_ian_su from superuser (i)      +| 
                       |                                               |                |                                          | u3_sam_su from superuser (s)       | 
 cr_admin              | Create Role                                   |                |                                          |                                    | 
 u3_amy_admin_su       | of g3_advisor_admin_su from superuser         |                |                                          |                                    | 
 u3_earl_su            | empty of g3_service_advisor_su from superuser |                |                                          |                                    | 
 u3_ian_admin_su       | of g3_advisor_admin_su from superuser         |                | g3_advisor_admin_su from superuser (i)   |                                    | 
 u3_ian_su             |                                               |                | g3_service_advisor_su from superuser (i) |                                    | 
 u3_sam_admin_su       | of g3_advisor_admin_su from superuser         |                | g3_advisor_admin_su from superuser (s)   |                                    | 
 u3_sam_su             |                                               |                | g3_service_advisor_su from superuser (s) |                                    | 
(9 rows)

